pkg/input/wordlist.go

```diff
@@ -117,17 +118,44 @@ func (w *WordlistInput) readFile(path string) error {
 					data = append(data, []byte(contnt))
 				}
 			} else {
-				data = append(data, []byte(reader.Text()))
+				text := reader.Text()
+
+				if w.config.IgnoreWordlistComments {
+					text = stripComments(text)
+				}
+				data = append(data, []byte(text))
 			}
 		} else {
-			data = append(data, []byte(reader.Text()))
+			text := reader.Text()
+
+			if w.config.IgnoreWordlistComments {
+				text = stripComments(text)
+			}
+			data = append(data, []byte(text))
 			if w.keyword == "FUZZ" && len(w.config.Extensions) > 0 {
 				for _, ext := range w.config.Extensions {
-					data = append(data, []byte(reader.Text()+ext))
+					data = append(data, []byte(text+ext))
 				}
 			}
 		}
 	}
 	w.data = data
 	return reader.Err()
 }
+
+// stripComments removes all kind of comments from the word
+func stripComments(text string) string {
+	// If the line starts with a # ignoring any space on the left,
+	// return blank.
+	if strings.HasPrefix(strings.TrimLeft(text, " "), "#") {
+		return ""
```

Comment:
Lines that start with comment should be ignored completely instead of adding an empty entry, while still letting the user to actually include an empty entry to the wordlist.

To handle this the `stripComments` should return `(string, error)` or `(string, bool)` to figure out if we should skip the entry in the caller function.

Reply:
Fixed.